tools/packaging/guest-image/build_image.sh

```diff
@@ -75,46 +68,81 @@ Usage:
 ${script_name} [options]
 
 Options:
- -v <version> : Kata version to build images. Use kata release for
-                 for agent and osbuilder.
-
+ --imagetype=${image_type}
+ --prefix=${prefix}
+ --destdir=${destdir}
 EOT
 
 	exit "${return_code}"
 }
 
 main() {
-	while getopts "v:h" opt; do
+	image_type=image
+	destdir="$PWD"
+	prefix="/opt/kata"
+	builddir="${PWD}"
+	while getopts "h-:" opt; do
 		case "$opt" in
+		-)
+			case "${OPTARG}" in
+			imagetype=image)
+				image_type=image
+				#image information
+				img_distro=$(get_from_kata_deps "assets.image.architecture.${arch_target}.name")
+				img_os_version=$(get_from_kata_deps "assets.image.architecture.${arch_target}.version")
+				image_name="kata-${img_distro}-${img_os_version}.${image_type}"
+				;;
+			imagetype=initrd)
+				image_type=initrd
+				#initrd information
+				initrd_distro=$(get_from_kata_deps "assets.initrd.architecture.${arch_target}.name")
+				initrd_os_version=$(get_from_kata_deps "assets.initrd.architecture.${arch_target}.version")
+				initrd_name="kata-${initrd_distro}-${initrd_os_version}.${image_type}"
+				;;
+			prefix=*)
+				prefix=${OPTARG#*=}
+				;;
+			destdir=*)
+				destdir=${OPTARG#*=}
+				;;
+			builddir=*)
+				builddir=${OPTARG#*=}
+				;;
+			*)
+				echo "Invalid option -$opt${OPTARG}"
+				usage 1
+				;;
+			esac
+			;;
 		h) usage 0 ;;
-		v) kata_version="${OPTARG}" ;;
 		*)
 			echo "Invalid option $opt"
 			usage 1
 			;;
 		esac
 	done
+	readonly destdir
+	readonly builddir
+
+	echo "build ${image_type}"
 
-	install_yq
 
-	#image information
-	img_distro=$(get_from_kata_deps "assets.image.architecture.${arch_target}.name" "${kata_version}")
-	#In old branches this is not defined, use a default
-	img_distro=${img_distro:-clearlinux}
-	img_os_version=$(get_from_kata_deps "assets.image.architecture.${arch_target}.version" "${kata_version}")
 
-	#initrd information
-	initrd_distro=$(get_from_kata_deps "assets.initrd.architecture.${arch_target}.name" "${kata_version}")
-	#In old branches this is not defined, use a default
-	initrd_distro=${initrd_distro:-alpine}
-	initrd_os_version=$(get_from_kata_deps "assets.image.architecture.${arch_target}.version" "${kata_version}")
+	install_dir="${destdir}/${prefix}/share/kata-containers/"
+	readonly install_dir
+
+	mkdir -p "${install_dir}"
 
-	shift "$((OPTIND - 1))"
 	pushd "${osbuilder_dir}"
-	build_initrd
-	build_image
-	create_tarball
-	cp "${tarball_name}" "${destdir}"
+	case "${image_type}" in
+	initrd)
+		build_initrd
+		;;
+	image)
+		build_image
+		;;
```

Comment:
Nit: you could collapse these values down to a single line:

```suggestion
	initrd) build_initrd ;;
	image) build_image ;;
```